community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStore.java

```diff
@@ -893,14 +907,18 @@ public long nextCommittingTransactionId()
     }
 
     @Override
-    public void transactionCommitted( long transactionId, long checksum )
+    public void transactionCommitted( long transactionId, long checksum, long logVersion, long byteOffset )
     {
-        if ( lastCommittedTx.offer( transactionId, checksum ) )
+        if ( lastCommittedTx.offer( transactionId, new long[]{checksum, logVersion, byteOffset} ) )
         {
             long[] transactionData = lastCommittedTx.get();
             setRecord( Position.LAST_TRANSACTION_ID, transactionData[0] );
             setRecord( Position.LAST_TRANSACTION_CHECKSUM, transactionData[1] );
+            setRecord( Position.LAST_TRANSACTION_LOG_VERSION, transactionData[2] );
+            setRecord( Position.LAST_TRANSACTION_LOG_BYTE_OFFSET, transactionData[3] );
             lastTransactionChecksum = checksum;
+            lastTransactionLogVersion = logVersion;
+            lastTransactionLogByteOffset = byteOffset;
```

Comment:
The boolean returned by `lastCommittedTx.offer()` means that only the transaction with the greatest transactionId will have its logVersion and byteOffset (along with the other data) published? Which should be fine because the transaction id is representative of the order of the transactions in the log because the BatchingTransactionAppender both produces the transaction id and appends the transaction to the log, in the same critical section locking the log file.

Yeah... I think this is good.